### DISCUSSION TITLE
How to setup properly a consumer prefetch count

### DISCUSSION THREAD

**Discussion by DavidSantacruzR** [gftea/amqprs#103] [Q&A] [answered]
Hi there, I've been playing around with the library, and recently started testing consumers with a prefetch count.

However, I haven't been able to set one properly as all the messages get acknowledged instantly.

Any advice or code example is much appreciated.

I've been trying code like this one.

```rust
       channel
            .basic_consume(DefaultConsumer::new(args.no_ack), args)
            .await
            .unwrap();

        channel
            .basic_qos(BasicQosArguments::new(0, 100, false))
            .await
            .unwrap();
```

**Comment by gftea** [comment]
You can not use autoack mode if want to set prefetch. Meanwhile you should be able to check effective prefetch count value on server side

**Reply by DavidSantacruzR** [reply, marked as answer]
I see the prefetch count set to 1 for example, but I when the consumer is running, I don't see it acknowledging messages one by one. I guess I need to do more research.
